mlflow/types/utils.py

```diff
@@ -478,3 +478,46 @@ def _infer_schema_from_type_hint(type_hint, examples=None):
     else:
         _logger.info("Unsupported type hint: %s, skipping schema inference", type_hint)
         return None
+
+
+def _infer_param_schema(parameters: Dict[str, Any]):
+    if not isinstance(parameters, dict):
+        raise MlflowException(
+            f"Expected parameters to be dict, got {type(parameters).__name__}",
+        )
+
+    def _infer_type_and_shape(value):
+        ndim = np.array(value).ndim
+        if ndim == 1:
+            if not isinstance(value, list):
+                raise MlflowException(
+                    f"Expected parameters to be 1D array or scalar, got {type(value).__name__}",
+                    INVALID_PARAMETER_VALUE,
+                )
+            shape = (-1,)
```

Comment:
I think we should also allow `np.array` / `np.ndarray` instances. Can we update this `isinstance` check?

Reply:
For example, the following seems like a perfectly valid use case:

```
import mlflow
from mlflow.models import infer_signature

import numpy as np

s = infer_signature(["k"], params={"foo": [8]})


class MyMod(mlflow.pyfunc.PythonModel):
    def predict(self, ctx, inp, params):
        return [7 for item in inp]


with mlflow.start_run():
    m = mlflow.pyfunc.log_model(
        python_model=MyMod(),
        artifact_path="foo",
        signature=s,
        pip_requirements=[],
    )

m = mlflow.pyfunc.load_model(m.model_uri)
print(m.predict(["k"], params={"foo": np.array([8])}))
```